akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -178,8 +178,19 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
             }
         }
         case h: `Transfer-Encoding` => teh match {
-          case None    => parseHeaderLines(input, lineEnd, headers, headerCount + 1, ch, clh, cth, Some(h), e100c, hh)
-          case Some(x) => parseHeaderLines(input, lineEnd, headers, headerCount, ch, clh, cth, Some(x append h.encodings), e100c, hh)
+          case None =>
+            h.encodings match {
+              case Seq(TransferEncodings.chunked) =>
+                // A single chunked is the only one we support
+                parseHeaderLines(input, lineEnd, headers, headerCount + 1, ch, clh, cth, Some(h), e100c, hh)
+              case Seq(unknown) =>
+                failMessageStart(s"HTTP unsupported Transfer-Encoding entry '${unknown.name}'")
```

Comment:
```suggestion
                failMessageStart(s"Unsupported Transfer-Encoding '${unknown.name}'")
```

Reply:
Needed changes in tests as well, fixed in latest commit

Reply:
Thanks